### PR TITLE
Add option to disable empty-element tags

### LINKED ISSUE #64: Option for disabling `OPEN_CLOSE_TAG`
Hiya! So the Apache parser doesn't support the "open/close tag" (AKA empty-element tag) -- that is, using `<block/>` as a shortcut for `<block> </block>`. This enables people to have a section like:
```
<Directory />
</Directory>
```
(a pretty common apache block!) without breaking the parser. Perl's Config::General throws an error when parsing the above since the first line is interpreted as an open/close tag, but Apache's parser is fine with it.

To resolve this conflict, I might propose adding an option to disable OPEN_CLOSE_TAG parsing.

### PR DESCRIPTION
**Pull request by sydneyli**
Fixes #64 -- disables self-closing tags. Falls back to tokenizing them as open tags, if possible.

This matches native Apache parser behavior.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -374,6 +374,13 @@ is `False`.
 
 Escaped newlines at the end of hash comments add the following line to the comment. Apache's native config parser processes hash comments this way. Set to `False` by default.
 
+### disableselfclosingtags
```
Comment (etingof) [inline README.md]: The other name for self-closing tags is "empty-element tag". Not sure which is better understood / more canonical.

Just noting this, I am good with any of these names.

File: apacheconfig/apacheconfigtool.py [hunk 2]
```diff
@@ -139,6 +139,11 @@ def main():
               'on code generation')
     )
 
+    options.add_argument(
+        '--disableselfclosingtags', action='store_true',
+        help=('Disables the parsing of empty, self-closing tags like <block/>')
```
Comment (etingof) [inline apacheconfig/apacheconfigtool.py]: nit: it's not an empty tag, but an empty element tag. ;-)

File: tests/unit/test_lexer.py [hunk 3]
```diff
@@ -119,6 +119,18 @@ def testBlockBlankLines(self):
...
         self.assertEqual(tokens, ['a', '\n\n\n', 'a', '\n'])
 
+    def testSelfClosingTags(self):
+        text = "<block/>"
+        tokens = self.lexer.tokenize(text)
+        self.assertEqual(tokens, ['block'])
+
+    def testSelfClosingTagsDisabled(self):
+        options = { 'disableselfclosingtags': True }
+        text = "<block/>\n<block />\n<block hello/>"
```
Comment (etingof) [inline tests/unit/test_lexer.py]: Just wondering, would `"""` block of text be more readable/manageable than a single line with embedded `\n`? ;-)

File: README.md (hunk 1 above)
Comment (sydneyli) [reply README.md]: Not sure either! I just did some digging and did find "empty-element tag" used in the [wikipedia page for XML](https://en.wikipedia.org/wiki/XML#Key_terminology) -- so let's go ahead with that.

**Comment by etingof** [comment]
Thank you!

**Pull request merged**